client/src/api/SSOTretrieval.js

```diff
@@ -24,7 +24,19 @@
       return response;
     };
 
+    /**
+     * @description Create a new robot with the specified name for the specified user 
+     * @param {String} newName - String including the user Id
+     */
+     const createNewBot = async (userid, newName) => {
+        const adjustedName = newName.replace(/\s/g, '+');
```

Comment:
what does this do?

Reply:
Exactly;
we need to replace all the spaces with another symbol for the communication to the server. That's what this regex does.